ee/WFLY-1160-Provide_ability_to_configure_a_global_directory_which_puts_the_contents_in_the_classpath.adoc

```diff
@@ -0,0 +1,137 @@
+  [standalone@localhost:9990 /] /subsystem=ee:list-add(name=global-modules,value={name=com.test})
+
+This configuration implies the following:
+
+  * Users have to specify all required resource roots using --absolute-resources argument.
+  * It is necessary to modify the generated module.xml, or at least re-create the global module if we want to add/remove any additional resource-root for this global module.
+  * This configuration allows any customization of the generated module.xml, making it very flexible to cover broader cases, for example, it is possible to declare module dependencies, mark them as exported, declare the main class and so on.
+  * This command is not appropriate for use in a managed domain or when connecting to the management CLI remotely. Modules should be added and removed manually in a production environment.
+
+This RFE is opened to have an alternative new operation to address simpler cases, where it is not required any additional configuration beyond on the default module.xml configurations, and supported in production environments and domain mode.
+
+The simplest way to accomplished this is to use the EE subsystem to configure a path to the designated directory that will contain the libraries we want to be shared across all deployed applications. A new resource named `global-directory` will be added to the EE subsystem. This resource will use two attributes path and relative-to to configure a global directory:
+
+  [standalone@localhost:9990 /] /subsystem=ee/global-directory=my-common-libs:add(path=lib, relative-to=jboss.home.dir)
+
+  [domain@localhost:9990 /] /profile=default/subsystem=ee/global-directory=my-common-libs:add(path=lib, relative-to=jboss.server.data.dir)
+
+The following attributes will be available on the global-directory resource:
+
+* path: The path of the directory to scan. (Mandatory)
+* relative-to: The name of another previously named path, or of one of the standard paths provided by the system. (Optional)
+
+When a new `global-directory` is created, the server will create a module that will have a https://jboss-modules.github.io/jboss-modules/manual/#path-resource-loader[Path Resource Loader] using 'path' and 'relative-to' attributes and one https://jboss-modules.github.io/jboss-modules/manual/#jar-resource-loader[Jar Resource loader] for each jar file included in this directory and subdirectories.
+
+The 'Path Resource Loader' will make available any file as a resource to the applications.
+The 'Jar Resource loader' will make available any class inside of the jar file to the applications.
+
+The jar resources of this module will be created iterating over all jar files in alphabetical order and size descending. The resource root order declaration will govern the class loading order for the classes available from the jar resources of this module.
```

Comment:
Hi @yersan, what if I have a/c.jar and d/b.jar. What is the first resource and what is the second? b.jar < c.jar? Or a/c.jar < d/b.jar?

Reply:
Hello @marekkopecky, Sorry, I missed out this notification.

Currently, we are only taking into account jar file names, not the full pathname. So in your example, b.jar will be before than c.jar

There is a similar question from Tomas Terem in the Wildfly PR, we can discuss it there:
https://github.com/wildfly/wildfly/pull/12489#issuecomment-521720014